pandas/core/dtypes/cast.py

```diff
@@ -146,6 +147,30 @@ def maybe_box_datetimelike(value: Scalar, dtype: Optional[Dtype] = None) -> Scal
     return value
 
 
+def maybe_box_native(value: "Union[Series, Scalar]") -> "Union[Series, Scalar]":
+    """
+    If passed a scalar cast the scalar to a python native type.
+
+    Parameters
+    ----------
+    value : scalar or Series
+
+    Returns
+    -------
+    scalar or Series
+    """
+    if not is_scalar(value):
+        pass
+    elif is_datetime_or_timedelta_dtype(value):
+        value = maybe_box_datetimelike(value)
+    elif is_float_dtype(value):
+        value = float(value)
+    elif is_integer_dtype(value):
+        with suppress(ValueError):
```

Comment:
bool?

Reply:
I had `np.bool_` in the test and somehow it passed without an explicit check here - but added one now